src/strands/multiagent/graph.py

```diff
@@ -578,6 +578,14 @@ async def _execute_node(self, node: GraphNode, invocation_state: dict[str, Any])
                     else:
                         agent_response = await node.executor.invoke_async(node_input, invocation_state=invocation_state)
 
+                    if agent_response.stop_reason == "interrupt":
+                        node.executor.messages.pop()  # remove interrupted tool use message
```

Comment:
The only way this would matter is if the message history was preserved between node-runs, which isn't true right?

E.g. this is more of a defensive check than anything?